src/NUnitFramework/framework/Internal/Randomizer.cs

```diff
@@ -82,15 +107,12 @@ public static Randomizer GetRandomizer(ParameterInfo parameter)
         /// <summary>
         /// Create a new Randomizer using the next seed
         /// available to ensure that each randomizer gives
-        /// a unique sequence of _values.
+        /// a unique sequence of values.
         /// </summary>
         /// <returns></returns>
         public static Randomizer CreateRandomizer()
         {
-            if (seedGenerator == null)
-                seedGenerator = new Random(InitialSeed);
-
-            return new Randomizer(seedGenerator.Next());
+            return new Randomizer(SeedGenerator.Next());
```

Comment:
Exactly. A Randomizer is used to generate the tests. Then each test method gets it's own randomizer to use for execution purposes. Those randomizers are kept in a dictionary so that the method uses the same randomizer no matter when it is run. Of course, if we decide to load assemblies in parallel, we'll have to deal with it then.